test/e2e/specs/editor/various/autocomplete-and-mentions.spec.js

```diff
@@ -0,0 +1,429 @@
+			await page.keyboard.type( testData.postCompleterInput );
+			// The characters before `Escape` should remain (i.e. `~app`)
+			expect( await editor.getEditedPostContent() ).toBe(
+				testData.snapshot
+			);
+		} );
+
+		// This test does not apply to user mentions, because they don't get disabled.
+		if ( type !== 'mention' ) {
+			test( `${ completer }: should not insert disabled ${ type }s`, async ( {
+				page,
+				editor,
+			} ) => {
+				await page.click( 'role=button[name="Add default block"i]' );
+				// The 'Grapes' option is disabled in our test plugin, so it should not insert the grapes emoji
+				await page.keyboard.type( 'Sorry, we are all out of ~g' );
+				await expect(
+					page.locator(
+						'//button[@role="option"][text()="🍇 Grapes"]'
+					)
+				).toBeVisible();
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( ' grapes.' );
+				expect( await editor.getEditedPostContent() )
+					.toBe( `<!-- wp:paragraph -->
+<p>Sorry, we are all out of ~g grapes.</p>
+<!-- /wp:paragraph -->` );
+			} );
+		}
+
+		test( `${ completer }: should allow newlines after multiple ${ type } completions`, async ( {
+			page,
+			editor,
+		} ) => {
+			const testData = {};
+			if ( type === 'mention' ) {
+				testData.triggerString = '@bu';
+				testData.optionPath = '//*[contains(text(),"Buddy Elf")]';
+				testData.snapshot = `<!-- wp:paragraph -->
+<p>@buddytheelf test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>@buddytheelf test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>@buddytheelf test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>@buddytheelf test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->`;
+			} else if ( type === 'option' ) {
+				testData.triggerString = '~b';
+				testData.optionPath = '[text()="🫐 Blueberry"]';
+				testData.snapshot = `<!-- wp:paragraph -->
+<p>🫐 test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>🫐 test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>🫐 test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>🫐 test</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->`;
+			}
+
+			await page.click( 'role=button[name="Add default block"i]' );
+
+			for ( let i = 0; i < 4; i++ ) {
+				await page.keyboard.type( testData.triggerString );
+				await expect(
+					page.locator(
+						`//button[@role="option"]${ testData.optionPath }`
+					)
+				).toBeVisible();
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( ' test' );
+				await page.keyboard.press( 'Enter' );
+			}
+
+			expect( await editor.getEditedPostContent() ).toBe(
+				testData.snapshot
+			);
+		} );
+	} );
+
+	// The following test attempts to cover an infinite loop regression (https://github.com/WordPress/gutenberg/issues/41709).
+	// Unfortunately, the regression (if present) crashes the tests, as well as the editor. If you've updated anything related to
```

Comment:
Do you know if Playwright will terminate this test if it timeouts?

Reply:
🥳 It _does!_ That's great news, I'm glad you thought to ask. I'm updating the code comment... now that we're in Playwright, it looks like this test does actually function the way we want it to!

Reply:
That's great!